source/js/searchRecipe.js

```diff
@@ -0,0 +1,48 @@
+function searchRecipe(key,apiKey){
+  let query = "https://api.spoonacular.com/recipes/complexSearch?apiKey="+apiKey+"&query="+key;
+  const recipeData = {}
+  fetch(query)
+  .then(response => response.json())
+  .then(data => {
+    for(let i=0;i<=data["results"].length;i++){
+      
+      let id = data["results"][i]["id"];
+      let title = data["results"][i]["title"];
+      let image = data["results"][i]["image"];
+      let description = 0;
+      let cost;
+      let ingredients = []
+        
+      fetch("https://api.spoonacular.com/recipes/"+id+"/information?apiKey="+apiKey)
+      .then(response => response.json())
+      .then(data => {
+        cost = data["pricePerServing"];
+        description = data["summary"];
+        let ingInfo = data["extendedIngredients"];
```

Comment:
Codacy has a fix for the issue: ['ingInfo' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/cse110-fa21-group7/cse110-fa21-group7/pullRequest?prid=8445062)
```suggestion
        const ingInfo = data["extendedIngredients"];
```